compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -165,6 +165,36 @@ CompareCondition reverseCondition(CompareCondition cond)
       }
    }
 
+/**
+ * \brief
+ *    Returns a condition that should be used if the order of operands to the compare instruction
+ *    is flipped.
+ *
+ * \param cond
+ *    The original condition.
+ *
+ * \return
+ *    The condition that should be used if the compare operands are flipped.
+ */
+CompareCondition flipConditionOrder(CompareCondition cond)
+   {
+   switch (cond)
+      {
+      case CompareCondition::eq:
+         return CompareCondition::eq;
+      case CompareCondition::ne:
+         return CompareCondition::ne;
+      case CompareCondition::lt:
+         return CompareCondition::gt;
+      case CompareCondition::ge:
+         return CompareCondition::le;
+      case CompareCondition::gt:
+         return CompareCondition::lt;
+      case CompareCondition::le:
+         return CompareCondition::ge;
+      }
```

Comment:
I'd like to see a fatal assert in the default case if this is meant to be an exhaustive switch. I don't recall if the compiler issues a warning or an error if not all cases are covered. If it's an error I suppose we can skip the fatal assert.

Reply:
Oops, looks like I'm getting too used to writing in Rust where all pattern matches are required to be exhaustive and enums are sane :sweat_smile:. Added d1e6b9efd5cb23660f29cb1f5216d5b8a2127462 to add default cases to a bunch of similar locations.